hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -24,9 +24,29 @@ type PropertyDefinition struct {
 	propertyType                     Type
 	description                      string
 	hasKubebuilderRequiredValidation bool
+	flatten                          bool           // maps to x-ms-client-flatten: should the propertyType be flattened into the parent
+	flattenedFrom                    []PropertyName // a list of property names from whence this property was flattened
```

Comment:
Elsewhere, you have in `buildFlattenedAssignment` a check like this:

``` go
if len(toProp.FlattenedFrom()) > 1 {
	// this doesn't appear to happen anywhere in the JSON schemas currently
	panic("need to implement multiple levels of flattening!")
}
```

Doesn't that mean that `flattenedFrom` will only ever have one member?

Reply:
At the moment. In the future some team might go crazy and squash multiple levels… it’s easier to have this do the right thing and then fail later in the pipeline.